packages/langium/src/validation/validation-registry.ts

```diff
@@ -36,6 +36,8 @@ export type ValidationAcceptor = <N extends AstNode>(severity: 'error' | 'warnin
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export type ValidationCheck = (node: any, accept: ValidationAcceptor, cancelToken: CancellationToken) => MaybePromise<void>;
 
+export type ValidationChecks<T extends string> = { [type in T]?: ValidationCheck | ValidationCheck[] }
```

Comment:
Please update all example languages as well as the Yeoman template to use this type.